doc/api/worker_threads.md

```diff
@@ -61,6 +61,113 @@ Worker threads inherit non-process-specific options by default. Refer to
 [`Worker constructor options`][] to know how to customize worker thread options,
 specifically `argv` and `execArgv` options.
 
+## `worker.connect(target[, data][, timeout])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+* `target` {number} The target thread id.
+* `data` {any} Any arbitrary, cloneable JavaScript value.
+  The data will be passed as the second argument to the callback provided in the
+  target thread via [`worker.setConnectionsListener()`][].
+* `timeout` {number} Time to wait for the communication port to the target thread,
+  in milliseconds. By default it's `undefined`, which means wait forever.
+* Returns: {Promise} A promise for a `MessagePort`.
+
+Establishes a connection to another worker thread in the same process, returning a
+`MessagePort` that can be used for the communication.
+
+The target thread must have a connection listener setup via [`worker.setConnectionsListener()`][]
+otherwise the connection request will fail.
+
+This method should be used when the target thread is not the direct parent or children of the current thread.
+The example below creates 10 nested threads, the last one will try to communicate with the third one.
```

Comment:
non-blocking suggestion: this could probably be expanded a bit to explain more about when someone should choose to use `connect(...)` vs. `worker.postMessage(...)` vs. `BroadcastChannel`. For instance, if someone has the `worker` instance, then using `connect(...)` is unnecessary because they can simply call `worker.postMessage(...)` to accomplish largely the same thing.

Reply:
I've added a note, thanks!